andela/Ah-backend-guardians
# #162685660 User receives email upon successful registration

#### What does this PR do?
This PR delivers a feature that enables a user to receive an email with an account activation upon successful registration.

#### Description of Task to be completed?
When a user suucessfuly signs up to the Authors Haven Application, he/she  should receive a verification email with a link to the application.

#### How should this be manually tested?

Before testing run these commands in cmd

`git clone https://github.com/andela/Ah-backend-guardians.git`

`cd Ah-backend-guardians`

`git checkout ft-email-confirmation-162685660`

`virtualenv env`

`. env/bin/activate`

`pip install -r requirements.txt`

1.  Create a postgres database and a `.env` file in your root directory and add the following configurations.
```
SECRET_KEY=andela_sjdndkjghdf78872

DB_NAME=your_database_name

DB_USER=database_user

DB_PASSWORD=database_password

DB_HOST=127.0.0.1

ALLOWED_HOSTS=.localhost, .herokuapp.com

EMAIL_PORT=587

EMAIL_HOST_USER = gauthorshaven@gmail.com

EMAIL_HOST_PASSWORD = guardiansauthors
```
2. Make sure an email account or if not create one. This account will receive the email activation link. 
3. Open Postman and type `http://localhost:8000/api/users/` in url text box.
4. In the body section of postman, type in the following json format. Note type in correct email address which will receive the link.
  {  
     "user":{
                    "email":"your_email_addresss",
                      "password": "your_password",
                      "username":"Greesn123d44ssfse"
                }
  }
5. Send a POST request to the url and then go to your email inbox to check wheter you have received the email.
6. click on the activation link to activate your account.

#### What are the relevant pivotal tracker stories?
#162685660

#### Screenshots (if appropriate)
![screen shot 2019-01-15 at 12 26 41 pm](https://user-images.githubusercontent.com/13882936/51171354-bec75500-18c1-11e9-8c56-89cc98e67711.png)
